integration-tests/src/test/java/org/zowe/apiml/integration/ha/GatewayMultipleInstancesTest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.integration.ha;
+
+import com.jayway.jsonpath.DocumentContext;
+import com.jayway.jsonpath.JsonPath;
+import io.restassured.RestAssured;
+import io.restassured.path.xml.XmlPath;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+
+import org.apache.http.util.EntityUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.zowe.apiml.util.categories.HATest;
+import org.zowe.apiml.util.config.ConfigReader;
+import org.zowe.apiml.util.config.GatewayServiceConfiguration;
+import org.zowe.apiml.util.http.HttpRequestUtils;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.core.Is.is;
+import static org.junit.jupiter.api.Assumptions.*;
+import static org.zowe.apiml.util.SecurityUtils.getConfiguredSslConfig;
+
+/**
+ * Verify that both Gateway instances are UP
+ */
+@HATest
+public class GatewayMultipleInstancesTest {
+    private GatewayServiceConfiguration gatewayServiceConfiguration;
+    private final String HEALTH_ENDPOINT = "/application/health";
+    private final String EUREKA_APPS = "/eureka/apps";
+    private String username;
+    private String password;
+    private int instances;
+
+    @BeforeEach
+    void setUp() {
+        gatewayServiceConfiguration = ConfigReader.environmentConfiguration().getGatewayServiceConfiguration();
+        username = ConfigReader.environmentConfiguration().getCredentials().getUser();
+        password = ConfigReader.environmentConfiguration().getCredentials().getPassword();
+        instances = gatewayServiceConfiguration.getInstances();
+    }
+
+    @Nested
+    class GivenMultipleGatewayInstances {
+        @Nested
+        class WhenSendingRequest {
+            @Test
+            void gatewayInstancesAreUp() throws IOException {
+                assumeTrue(instances > 1);
+                RestAssured.config = RestAssured.config().sslConfig(getConfiguredSslConfig());
+                assertThat(gatewayServiceConfiguration.getInternalPorts(), is(not(nullValue())), is(not("")));
+
+                checkInstancesAreUp();
+            }
+
+            @Test
+            void gatewayInstancesAreRegistered() throws URISyntaxException {
+                assumeTrue(instances > 1);
+                //@formatter:off
+                String xml =
+                    given()
+                        .auth().basic(username, password)
```

Comment:
There is already a piece within DiscoveryRequests class wouldn't it be better to adapt it and not replicate? It's also using the SSL config

Reply:
I was not aware of this class, yes we can use it, it makes sense

Reply:
Thanks